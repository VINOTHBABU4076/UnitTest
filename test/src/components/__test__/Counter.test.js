import {render,fireEvent,screen} from '@testing-library/react'  
import  renderer   from 'react-test-renderer'


import Counter from '../Counter'

 describe('Counter', () =>{
    it('should increment Counter', () =>{


     render(<Counter/>)

     const counter= screen.getByTestId('counter')
     const incrementbtn=screen.getByTestId('increment')

     fireEvent.click(incrementbtn)

     expect(counter).toHaveTextContent('2')
 })
 it('should decrement Counter',()=>{
    render(<Counter/>)

     const counter= screen.getByTestId('counter')
     const decrementbtn=screen.getByTestId('decrement')

     fireEvent.click(decrementbtn)

     expect(counter).toHaveTextContent('0')
 })

 })
describe('Counter Snapshot',() =>{
    it('should matches DOM Snapshot',() => {
        const tree=renderer.create(<Counter/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
 })

