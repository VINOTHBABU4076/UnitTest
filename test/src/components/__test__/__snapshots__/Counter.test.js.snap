// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot should matches DOM Snapshot 1`] = `
<div>
  <button
    data-testid="increment"
    onClick={[Function]}
  >
    increase
  </button>
  <p
    data-testid="counter"
  >
    count: 
    1
  </p>
  <button
    data-testid="decrement"
    onClick={[Function]}
  >
    decrease
  </button>
</div>
`;
